homeassistant/components/lcn.py

```diff
@@ -0,0 +1,200 @@
+
+DOMAIN = 'lcn'
+DATA_LCN = 'lcn'
+DEFAULT_NAME = 'pchk'
+
+CONF_SK_NUM_TRIES = 'sk_num_tries'
+CONF_DIM_MODE = 'dim_mode'
+CONF_OUTPUT = 'output'
+CONF_TRANSITION = 'transition'
+CONF_DIMMABLE = 'dimmable'
+CONF_CONNECTIONS = 'connections'
+
+DIM_MODES = ['steps50', 'steps200']
+OUTPUT_PORTS = ['output1', 'output2', 'output3', 'output4']
+
+# Regex for address validation
+PATTERN_ADDRESS = re.compile('^((?P<conn_id>\\w+)\\.)?s?(?P<seg_id>\\d+)'
+                             '\\.(?P<type>m|g)?(?P<id>\\d+)$')
+
+
+def has_unique_connection_names(connections):
+    """Validate that all connection names are unique.
+
+    Use 'pchk' as default connection_name (or add a numeric suffix if
+    pchk' is already in use.
+    """
+    for suffix, connection in enumerate(connections):
+        connection_name = connection.get(CONF_NAME)
+        if connection_name is None:
+            if suffix == 0:
+                connection[CONF_NAME] = DEFAULT_NAME
+            else:
+                connection[CONF_NAME] = '{}{:d}'.format(DEFAULT_NAME, suffix)
+
+    schema = vol.Schema(vol.Unique())
+    schema([connection.get(CONF_NAME) for connection in connections])
+    return connections
+
+
+def is_address(value):
+    """Validate the given address string.
+
+    Examples for S000M005 at myhome:
+        myhome.s000.m005
+        myhome.s0.m5
+        myhome.0.5    ("m" is implicit if missing)
+
+    Examples for s000g011
+        myhome.0.g11
+        myhome.s0.g11
+    """
+    matcher = PATTERN_ADDRESS.match(value)
+    if matcher:
+        is_group = (matcher.group('type') == 'g')
+        addr = (int(matcher.group('seg_id')),
+                int(matcher.group('id')),
+                is_group)
+        conn_id = matcher.group('conn_id')
+        return addr, conn_id
+    raise vol.error.Invalid('Not a valid address string.')
+
+
+LIGHTS_SCHEMA = vol.Schema({
+    vol.Required(CONF_NAME): cv.string,
+    vol.Required(CONF_ADDRESS): is_address,
+    vol.Required(CONF_OUTPUT): vol.All(vol.In(OUTPUT_PORTS), vol.Upper),
+    vol.Optional(CONF_DIMMABLE, default=False): vol.Coerce(bool),
+    vol.Optional(CONF_TRANSITION, default=0):
+        vol.All(vol.Coerce(float), vol.Range(min=0., max=486.),
+                lambda value: value * 1000),
+})
+
+CONNECTION_SCHEMA = vol.Schema({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Required(CONF_PORT): cv.port,
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Optional(CONF_SK_NUM_TRIES, default=3): cv.positive_int,
+    vol.Optional(CONF_DIM_MODE, default='steps50'): vol.All(vol.In(DIM_MODES),
+                                                            vol.Upper),
+    vol.Optional(CONF_NAME): cv.string
+})
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_CONNECTIONS): vol.All(
+            cv.ensure_list, has_unique_connection_names, [CONNECTION_SCHEMA]),
+        vol.Optional(CONF_LIGHTS, default=[]): vol.All(
```

Comment:
Maybe we should make this required since the component without lights is useless?